src/drawing/conics.rs

```diff
@@ -197,11 +197,11 @@ where
 {
     let mut x = radius;
     let mut y = 0i32;
-    let mut err = 0i32;
+    let mut p = 1 - radius;
     let x0 = center.0;
     let y0 = center.1;
 
-    while x >= y {
+    while x <= y {
```

Comment:
The initial value of `x` is larger than the initial value of `y` so it looks like this loop will never get executed.

Reply:
Fixed in the new commit.